src/webgpu/util/texture/texelData.ts

```diff
@@ -0,0 +1,359 @@
+      B: { bitLength, write },
+      A: { bitLength, write },
+    };
+
+    return new TexelDataRepresentationImpl(format, componentOrder, componentInfo, sRGB);
+  }
+
+  private totalBitLength(): number {
+    return this.componentOrder.reduce((acc, curr) => {
+      return acc + this.componentInfo[curr]!.bitLength;
+    }, 0);
+  }
+
+  private setComponent(data: ArrayBuffer, component: TexelComponent, n?: number): void {
+    const componentIndex = this.componentOrder.indexOf(component);
+    assert(componentIndex !== -1);
+    const bitOffset = this.componentOrder.slice(0, componentIndex).reduce((acc, curr) => {
+      return acc + this.componentInfo[curr]!.bitLength;
+    }, 0);
+    const bitLength = this.componentInfo[component]!.bitLength;
+
+    const write = this.componentInfo[component]!.write;
+    if (write === undefined) {
+      // Ignore components that are not writable (packed depth).
+      return;
+    }
+    assert(n !== undefined);
+    const { value, type } = write(n);
+    switch (type) {
+      case TexelWriteType.Float: {
+        const byteOffset = Math.floor(bitOffset / 8);
+        const byteLength = Math.ceil(bitLength / 8);
+        assert(byteOffset === bitOffset / 8 && byteLength === bitLength / 8);
+        switch (byteLength) {
+          case 4:
+            new DataView(data, byteOffset, byteLength).setFloat32(0, value, true);
+            break;
+          default:
+            unreachable();
+        }
+        break;
+      }
+      case TexelWriteType.Sint: {
+        const byteOffset = Math.floor(bitOffset / 8);
+        const byteLength = Math.ceil(bitLength / 8);
+        assert(byteOffset === bitOffset / 8 && byteLength === bitLength / 8);
+        switch (byteLength) {
+          case 1:
+            new DataView(data, byteOffset, byteLength).setInt8(0, value);
+            break;
+          case 2:
+            new DataView(data, byteOffset, byteLength).setInt16(0, value, true);
+            break;
+          case 4:
+            new DataView(data, byteOffset, byteLength).setInt32(0, value, true);
+            break;
+          default:
+            unreachable();
+        }
+        break;
+      }
+      case TexelWriteType.Uint: {
+        const byteOffset = Math.floor(bitOffset / 8);
+        const byteLength = Math.ceil(bitLength / 8);
+        if (byteOffset === bitOffset / 8 && byteLength === bitLength / 8) {
+          switch (byteLength) {
+            case 1:
+              new DataView(data, byteOffset, byteLength).setUint8(0, value);
+              break;
+            case 2:
+              new DataView(data, byteOffset, byteLength).setUint16(0, value, true);
+              break;
+            case 4:
+              new DataView(data, byteOffset, byteLength).setUint32(0, value, true);
+              break;
+            default:
+              unreachable();
+          }
+        } else {
+          // Packed representations are all 32-bit and use Uint as the data type.
```

Comment:
I don't think the layout of depth24plus formats are observable at all. If this branch isn't hit, maybe we should unreachable() for now.

Reply:
This is hit for rg10b11float and rgb10a2unorm

Reply:
Right.